docs/sources/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-log-queries.md

```diff
@@ -0,0 +1,508 @@
+---
+aliases:
+  - ../../../plugins/add-support-for-log-queries/
+description: Add features to log queries in Explore.
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - queries
+  - explore queries
+  - explore
+  - logs
+  - log
+labels:
+  products:
+    - enterprise
+    - oss
+title: Add features to log queries in Explore
+weight: 400
+---
+
+# Add features to log queries in Explore
+
+[Explore]({{< relref "../../../../explore" >}}) provides an excellent place for investigating incidents and troubleshooting using logs. If the data source produces log results, we highly recommend implementing the following APIs to empower your users to fully utilize the logs UI and its features within Explore.
+
+This guide will walk you through the process of adding support for Explore features in a data source plugin through a seamless integration, maximizing the potential for logs analysis. By implementing these APIs, you can enhance the user experience and provide valuable insights through Explore's powerful log investigation capabilities.
+
+## Supported APIs for external plugin developers
+
+### Logs data frame format
+
+Here is the proposal for how should data frames for log results look like: https://grafana.github.io/dataplane/logs/.
+
+The data frame should include following fields:
```

Comment:
Is it exactly the same as the description in data plane page? Maybe it's better not to repeat it, otherwise might be hard to keep it in sync. What do you think?

Reply:
Good point! Although data plane documentation is currently not published and only part of https://github.com/grafana/dataplane. So when it is published, we can remove this section. 

Also now thinking about it, I will delete the link as it is not an official link.